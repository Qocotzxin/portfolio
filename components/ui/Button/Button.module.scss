@use "@styles/variables" as vars;

.Button {
  width: fit-content;
  padding: vars.$ce-space-1;
  background: transparent;
  color: vars.$ce-color-blue-400;
  border-radius: vars.$ce-border-radius-md;
  transition: background 0.2s;

  &:hover {
    background: rgba(66, 153, 225, 0.1);
  }

  &--dark {
    color: vars.$ce-color-blue-200;

    &:hover {
      background: rgba(144, 205, 244, 0.1);
    }
  }

  & > svg {
    width: 24px;
    height: 24px;
  }
}
