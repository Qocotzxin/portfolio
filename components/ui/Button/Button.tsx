import { useIsDarkMode } from "@hooks/useIsDarkMode";
import { applyConditionalClass, concatClassNames } from "@utils/classnames";
import { ComponentPropsWithoutRef, forwardRef, PropsWithChildren } from "react";
import styles from "./Button.module.scss";

export type ButtonProps = PropsWithChildren<
  {
    isFullWidth?: boolean;
  } & ComponentPropsWithoutRef<"button">
>;

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, isFullWidth, ...buttonProps }, ref) => {
    const isDark = useIsDarkMode();

    return (
      <button
        className={concatClassNames(
          styles.Button,
          applyConditionalClass(isDark, styles["Button--dark"])
        )}
        {...buttonProps}
        ref={ref}
      >
        {children}
      </button>
    );
  }
);

Button.displayName = "Button";

export default Button;
