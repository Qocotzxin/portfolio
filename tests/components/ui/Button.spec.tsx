import { render } from "@testing-library/react";
import Button from "@ui/Button";

describe("Button", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<Button>Test</Button>);
    expect(container).toMatchSnapshot();
  });
});
