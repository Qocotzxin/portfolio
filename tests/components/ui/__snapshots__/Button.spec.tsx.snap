// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot. 1`] = `
<div>
  <button
    class="Button "
  >
    Test
  </button>
</div>
`;
